encoder/src/main/scala/org/clulab/scala_transformers/encoder/Encoder.scala

```diff
@@ -19,6 +20,19 @@ class Encoder(val encoderEnvironment: OrtEnvironment, val encoderSession: OrtSes
 
     val result: OrtSession.Result = encoderSession.run(inputs)
     val outputs = Math.fromResult(result)
+
+    if(nonLin.isDefined) {
+      for (matrix <- outputs) {
+        for (i <- 0 until Math.rows(matrix)) {
+          val row = Math.row(matrix, i)
+          for (j <- 0 until Math.cols(matrix)) {
+            val orig = Math.get(row, j)
+            Math.set(row, j, nonLin.get.compute(orig))
```

Comment:
If this does stay, the `nonLin.get` might be moved outside the triply nested loops.